src/IMetricPusher.cs

```diff
@@ -1,11 +1,12 @@
+using System;
 using System.Threading.Tasks;
 
 namespace Prometheus.Client.MetricPusher
 {
     /// <summary>
     ///     Service for Push Gateway
     /// </summary>
-    public interface IMetricPusher
+    public interface IMetricPusher : IDisposable
```

Comment:
I believe that `IMetricPusher` should not force you make disposable implementation.

```suggestion
    public interface IMetricPusher
```

Reply:
But there is a case when MetricPusher creates HttpClient, which I believe should be disposed as soon as possible.

Reply:
It is implementation creates `HttpClient`, not interface.